packages/components/src/color-picker/test/index.tsx

```diff
@@ -73,99 +38,132 @@ const legacyColorMatcher = {
+			expect( formatSelector ).toBeVisible();
 
-			expect( saturation ).toBeInTheDocument();
+			await user.selectOptions( formatSelector, 'hex' );
 
-			moveReactColorfulSlider(
-				saturation,
-				{ pageX: 0, pageY: 0 },
-				{ pageX: 10, pageY: 10 }
-			);
+			const hexInput = screen.getByRole( 'textbox' );
+			expect( hexInput ).toBeVisible();
 
-			await waitFor( () =>
-				expect( onChangeComplete ).toHaveBeenCalled()
-			);
+			await user.clear( hexInput );
+			await user.type( hexInput, '1ab' );
 
-			expect( onChangeComplete ).toHaveBeenCalledWith(
+			expect( onChangeComplete ).toHaveBeenCalledTimes( 3 );
+			expect( onChangeComplete ).toHaveBeenLastCalledWith(
 				legacyColorMatcher
 			);
 		} );
 	} );
+	describe( 'Hex input', () => {
+		it( 'should fire onChange with the correct value from the hex input', async () => {
+			const user = userEvent.setup();
+			const onChange = jest.fn();
+			const color = '#000';
+
+			render(
+				<ColorPicker
+					onChange={ onChange }
+					color={ color }
+					enableAlpha={ false }
+				/>
+			);
+
+			const formatSelector = screen.getByRole( 'combobox' );
+			expect( formatSelector ).toBeVisible();
+
+			await user.selectOptions( formatSelector, 'hex' );
 
-	it( 'should fire onChange with the string value', async () => {
-		const onChange = jest.fn();
-		const color = 'rgba(1, 1, 1, 0.5)';
+			const hexInput = screen.getByRole( 'textbox' );
+			expect( hexInput ).toBeVisible();
 
-		const { container } = render(
-			<ColorPicker onChange={ onChange } color={ color } enableAlpha />
-		);
+			await user.clear( hexInput );
+			await user.type( hexInput, '1ab' );
 
-		const saturation = getSaturation( container );
+			expect( onChange ).toHaveBeenCalledTimes( 3 );
+			expect( onChange ).toHaveBeenLastCalledWith( '#11aabb' );
+		} );
+	} );
 
-		if ( saturation === null ) {
-			throw new Error( 'The saturation slider could not be found' );
-		}
+	describe.each( [
+		[ 'red', 0, '#7dffff' ],
+		[ 'green', 1, '#ff7dff' ],
+		[ 'blue', 2, '#ffff7d' ],
+	] )( 'RGB inputs', ( colorInput, inputIndex, expected ) => {
+		it( `should fire onChange with the correct value when the ${ colorInput } value is updated`, async () => {
+			const user = userEvent.setup();
+			const onChange = jest.fn();
+			const color = '#fff';
 
-		expect( saturation ).toBeInTheDocument();
+			render(
+				<ColorPicker
+					onChange={ onChange }
+					color={ color }
+					enableAlpha={ false }
+				/>
+			);
 
-		moveReactColorfulSlider(
-			saturation,
-			{ pageX: 0, pageY: 0 },
-			{ pageX: 10, pageY: 10 }
-		);
+			const formatSelector = screen.getByRole( 'combobox' );
+			expect( formatSelector ).toBeVisible();
 
-		await waitFor( () => expect( onChange ).toHaveBeenCalled() );
+			await user.selectOptions( formatSelector, 'rgb' );
 
-		expect( onChange ).toHaveBeenCalledWith(
-			expect.stringMatching( /^#([a-fA-F0-9]{8})$/ )
-		);
-	} );
+			const inputElement =
+				screen.getAllByRole( 'spinbutton' )[ inputIndex ];
```

Comment:
These `spinbutton` elements have an accessible label, could be found more directly with `screen.getByRole( 'spinbutton', { name: 'Red' } )`. Instead of the `inputIndex`. The HSL inputs have labels, too. "Hue", "Saturation", etc., always with leading uppercase letter.

Reply:
Excellent point, thank you for pointing those out! Updated in 5363ad7677